### PR TITLE
[CELEBORN-83][FOLLOWUP] Fix various bugs when using HDFS as storage.

### PR DESCRIPTION
**Pull request by FMX**
# [CELEBORN-83][FOLLOWUP] Fix various bugs when using HDFS as storage.

### What changes were proposed in this pull request?
1.Fix class cast exception.
2.Fix clean hdfs expired app dir.
3.Fix npe when destroy file writers.
4.Adjust disk not usable log level.
5.Close hdfs output streams.
6.Change hdfs log level to reduce log size.
7.Add instructions for buffer size using HDFS.
8.Fix data loss when reading from HDFS.

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java [hunk 1]
```diff
@@ -98,6 +98,17 @@ public DfsPartitionReader(
           ShuffleClient.getHdfsFs(conf).open(new Path(location.getStorageInfo().getFilePath()));
       chunkOffsets.addAll(getChunkOffsetsFromUnsortedIndex(conf, location));
     }
+    StringBuilder builder = new StringBuilder();
+    for (Long offset : chunkOffsets) {
+      builder.append(offset + " ");
+    }
+    logger.debug(
```
Comment (RexXiong) [inline client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java:105]: use logger.isDebugEnabled, if true then put concat chunkOffsets and logger.debug into it's logic scope, but why not print offset list directly?

File: client/src/main/java/org/apache/celeborn/client/ShuffleClient.java [hunk 2]
```diff
@@ -100,6 +100,7 @@ public static FileSystem getHdfsFs(CelebornConf conf) {
       synchronized (ShuffleClient.class) {
         if (null == hdfsFs) {
           Configuration hdfsConfiguration = new Configuration();
+          hdfsConfiguration.set("fs.hdfs.impl.disable.cache", "false");
```
Comment (RexXiong) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClient.java:103]: suggest add some comments for this configuration

File: client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java (hunk 1 above)
Comment (FMX) [reply client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java]: OK.

File: client/src/main/java/org/apache/celeborn/client/ShuffleClient.java (hunk 2 above)
Comment (FMX) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClient.java:104]: OK.

**Pull request closed without merging**